### PR TITLE
Backport ES patches

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java [hunk 1]
```diff
@@ -245,6 +272,54 @@ public Entry withShards(ImmutableOpenMap<ShardId, ShardSnapshotStatus> shards) {
             );
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:272]: `withShards` was dropped in https://github.com/elastic/elasticsearch/commit/e91936512aa464593fa745f8234d48e6a54d50ca.
Seems like we can drop it as well, no usages in this branch.

Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:272]: That's unrelated to the patches here?

Comment (BaurzhanSakhariev) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:272]: I think it's related to the third commit - at least this method was [removed](https://github.com/elastic/elasticsearch/commit/e91936512aa464593fa745f8234d48e6a54d50ca#diff-73ee7c916310b19deb06e97d8f66e7da737f1dff22c93bfced5b1e7fa071855dL211) in the original patch being ported

Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java]: Removed it

**Review by BaurzhanSakhariev — APPROVED** [review]
👍

**Pull request merged**